### PR TITLE
fix: openid claim missing error

### PR DESCRIPTION
**Pull request by donatello**
## Description
The error implies an expected claim is missing even when the claim is present.
Added an additional error message to clarify the problem.

## Motivation and Context
Improve error messages to debug external IDP issues.

## How to test this PR?
Configure an external OIDC IDP without attaching a policy but with the expected custom claim present.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

**Review by kannappanr — APPROVED** [review]
LGTM

File: cmd/sts-handlers.go [hunk 1]
```diff
@@ -343,14 +343,21 @@ func (sts *stsAPIHandlers) AssumeRoleWithSSO(w http.ResponseWriter, r *http.Requ
 	// JWT custom claims.
 	var policyName string
 	policySet, ok := iampolicy.GetPoliciesFromClaims(m, iamPolicyClaimNameOpenID())
+	policies := strings.Join(policySet.ToSlice(), ",")
```
Comment (harshavardhana) [inline cmd/sts-handlers.go:346]: Not sure why this was moved here @donatello?

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/sts-handlers.go (hunk 1 above)
Comment (donatello) [reply cmd/sts-handlers.go:346]: Moved? It's a new variable. Just to reuse the slicing and joining work.

Comment (donatello) [reply cmd/sts-handlers.go:346]: It's reused in the error message.

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**